indexing-service/src/main/java/org/apache/druid/indexing/common/task/HadoopTask.java

```diff
@@ -48,13 +49,11 @@
 public abstract class HadoopTask extends AbstractBatchIndexTask
 {
   private static final Logger log = new Logger(HadoopTask.class);
-  private static final ExtensionsConfig EXTENSIONS_CONFIG;
-
-  static final Injector INJECTOR = GuiceInjectors.makeStartupInjector();
 
-  static {
-    EXTENSIONS_CONFIG = INJECTOR.getInstance(ExtensionsConfig.class);
-  }
+  // Note: static variables mean that this task cannot run in a shared JVM,
```

Comment:
The issue here is more basic: static variables. If task A and task B both run in the Indexer, then they share a single copy of the static variable (unless the Indexer loads them in isolated class loaders.) By contrast, the `ExtensionsLoader` was converted from static variables to a Guice-injected singleton to avoid this issue. For `ExtensionsLoader` the only time two instances will appear in the same JVM is in tests, which used to have special code to undo the static variables between tests.

The `HadoopTask` approach seems more like a bug than a feature. A good enhancement would be to modify the `HadoopTask` to use Guice (and the `ExtensionsLoader`, perhaps modified for the Hadoop dependencies) to avoid the static variable issue.

Reply:
Yeah, the HadoopTask is done in a weird way. But I was asking if anything bad happens due to sharing the statics between different tasks. The comment makes it sound like something will break, but I can't tell what it is.

Reply:
Looking more closely, you may be right that this particular use is OK. There will be a single injector and set of extensions used by multiple tasks, which _should_ be OK. Removed the comment.